### ISSUE TITLE
Travis is failing on Windows

### ISSUE THREAD

**Issue by sarthakpati** [CBICA/CaPTk#232]
**Describe the bug**
The error is the following message, which results in the build to fail:
> ITK source code directory path length is too long (54 > 50). Please move the ITK source code directory to a directory with a shorter path.
For instance, on my fork, the ITK build path is `C:/Users/travis/build/sarthakpati/CaPTk/bin/ITK-build`. This is a Windows-specific issue.

**To Reproduce**
Head over the travis-ci.com and look at a recent Windows build.

**Expected behavior**
The build should succeed.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: All

**Additional context**

**Comment by surrsurus** [comment]
So, this is entirely dependent on username length. For me I get 52 > 50, however I fixed this issue by simply compiling directly in ${TRAVIS_BUILD_DIR}, fixes the issue for me and should also work on the upstream but now the build hits 2 hours and never builds CaPTk because caching on Windows is broken.

**Comment by sarthakpati** [comment]
Oh man... I guess we can't do multi threaded builds there either?

**Comment by surrsurus** [comment]
Well, we only have two threads per instance, but regardless VTK takes a really long time to build, pretty much the number one contender as to why travis takes so long. On top of that Phuc and I are noticing strange things happening with ccache. Sometimes it optimizes a good bit and other times VTK suddenly needs an entire recompile. In addition ccache doesn't seem like its working for windows, but I'm still looking into it.

**Comment by sarthakpati** [comment]
I think we need a better mechanism for the cross compile builds. Do you know what the build time limit is for AppVeyor? I know that azure does not have any limits. Regardless, I think getting one Linux build (including regression testing) running properly on Travis is important before we explore other options more closely.

**Comment by surrsurus** [comment]
From the docs (https://www.appveyor.com/docs/build-configuration/):

```
Time limitations

All plans have 60 minutes quota per build job.
```

**Comment by sarthakpati** [comment]
Then that's out, for sure. Anyway, one step at a time. Let's wrap up the regression testing and then we can look at other options.

**Comment by sarthakpati** [comment]
Hopefully will be fixed with migration to Azure Pipelines #360

**Comment by sarthakpati** [comment]
Closing this issue since we will focus on Pipelines for this.
